python/Dockerfile

```diff
@@ -32,43 +34,89 @@ RUN apt-get update \
     tk-dev \
     xz-utils \
     zlib1g-dev \
+    # jq isn't necessary to build Python from source, it's used below for locating the download URLs
+    jq \
   && rm -rf /var/lib/apt/lists/*
 
 COPY --chown=dependabot:dependabot python/helpers /opt/python/helpers
 USER root
+# TODO: Now that switched from `pyenv install` which compiled from source to downloading / copying a pre-compiled python
+# we could entirely drop pyenv if we change our ruby code that calls `pyenv exec` to track which version of python to
+# call and uses the full python paths.
```

Comment:
A quick grep of `pyenv ` commands showed this actually should be pretty straightforward, since we already have to keep track of when to tell `pyenv` to switch to a different python version.

Dropping `pyenv` completely would simplify the Dockerfile and be one less dep we pull in. But again, rather handle that in a separate PR after letting this one marinate a bit.